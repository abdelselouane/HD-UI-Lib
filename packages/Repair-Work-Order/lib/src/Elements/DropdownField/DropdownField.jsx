import React from 'react';
import PropTypes from 'prop-types';

const DropdownField = ({ id, label, selectedItem, items, handleSelectedItem, dropdownOpen }) => (
<React.Fragment>
<div className={`dropdown-list ${dropdownOpen && "dropdown-opened"}`} 
  id={id}
  style={{ marginTop: '22px'}}
>
  <label>{label}</label>
  <div className="dropdown-label"
    tabIndex="0"
    // onClick={handleDropdownOpen}
  >
    {selectedItem.option}
  </div>
  <div className="dropdown-outer-list">
    <ul className="dropdown-inner-list">
      {
        items.map( (item, index) =>
          (<li
              aria-selected={selectedItem.id === index} 
              className="dropdown-list-item" 
              id={item.value} 
              key={index}
              onClick={ handleSelectedItem }
              tabIndex="-1"
              title={item.option}
              value={item.id}
            >
            {item.option}
          </li>)
        )
      }
    </ul>
  </div>
</div>

</React.Fragment>);

DropdownField.prototype = {
  id: PropTypes.string.isRequired,
  label: PropTypes.string,
  dropdownOpen: PropTypes.bool.isRequired,
  selectedItem: PropTypes.shape({
    id: PropTypes.number.isRequired,
    value: PropTypes.string,
    option: PropTypes.string.isRequired
  }).isRequired,
  items: PropTypes.shape({
    id: PropTypes.number.isRequired,
    value: PropTypes.string,
    option: PropTypes.string.isRequired
  }).isRequired,
  handleSelectedItem: PropTypes.func.isRequired,
  handleDropdownOpen: PropTypes.func.isRequired
};

export default DropdownField;