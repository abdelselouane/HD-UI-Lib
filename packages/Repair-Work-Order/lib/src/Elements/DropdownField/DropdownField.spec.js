import React from 'react';
import { shallow, mount } from 'enzyme';
import DropdownField from './DropdownField';
import toJson from 'enzyme-to-json';
import * as Enzyme from 'enzyme';
import ReactSixteenAdapter from 'enzyme-adapter-react-16';

Enzyme.configure({ adapter: new ReactSixteenAdapter() });


describe('<DropdownField />', () => {
    let wrapper;
    let props;

    beforeAll( () => {

        props = {
            id: 'dropDown',//Required
            label: '', //Optional
            dropdownOpen: false,
            selectedItem: {
                id: 1,
                value: 'Item1',
                option: 'Item1'
            },
            //Required Object[Item]: Should be set in the state
            handleSelectedItem: jest.fn(),
            //Required Func: Should be set in parent component
            handleDropdownOpen: jest.fn().mockImplementation( () => {
                props.dropdownOpen = true
            }),
            //Required Func: Should be set in parent component
            items: [{//Required
                id: 0,
                value: '',
                option: 'Work Order'
            },
            {
                id: 1,
                value: 'Item1',
                option: 'Item1'
            },
            {
                id: 2,
                value: 'Item2',
                option: 'Item2'
            },
            {
                id: 3,
                value: 'Item3',
                option: 'Item3'
            }]
        }
        wrapper = mount(<DropdownField  { ...props } />);
    });
    
    it('should render properly', () => {
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    // it('should open dropdown when dropdownOpen is truthful', () => {
    //     const dropdown = wrapper.find('div.dropdown-list');
    //     expect(dropdown).toHaveLength(1);

    //     const dropdownLabel = wrapper.find('div.dropdown-label');
    //     const spy = jest.spyOn(props, 'handleDropdownOpen');
    //     dropdownLabel.simulate('click');

    //     expect(spy).toHaveBeenCalled();
    //     expect(props.dropdownOpen).toBe(true);

    //     wrapper = mount(<DropdownField  { ...props } />);
    //     const dropdownOpened = wrapper.find('div.dropdown-opened');
    //     expect(dropdownOpened).toHaveLength(1);
    // })

});

