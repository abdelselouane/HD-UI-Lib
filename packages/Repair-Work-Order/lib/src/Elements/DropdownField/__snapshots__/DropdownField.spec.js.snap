// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DropdownField /> should render properly 1`] = `
<DropdownField
  dropdownOpen={false}
  handleDropdownOpen={[MockFunction]}
  handleSelectedItem={[MockFunction]}
  id="dropDown"
  items={
    Array [
      Object {
        "id": 0,
        "option": "Work Order",
        "value": "",
      },
      Object {
        "id": 1,
        "option": "Item1",
        "value": "Item1",
      },
      Object {
        "id": 2,
        "option": "Item2",
        "value": "Item2",
      },
      Object {
        "id": 3,
        "option": "Item3",
        "value": "Item3",
      },
    ]
  }
  label=""
  selectedItem={
    Object {
      "id": 1,
      "option": "Item1",
      "value": "Item1",
    }
  }
>
  <div
    className="dropdown-list false"
    id="dropDown"
    style={
      Object {
        "marginTop": "22px",
      }
    }
  >
    <label />
    <div
      className="dropdown-label"
      tabIndex="0"
    >
      Item1
    </div>
    <div
      className="dropdown-outer-list"
    >
      <ul
        className="dropdown-inner-list"
      >
        <li
          aria-selected={false}
          className="dropdown-list-item"
          id=""
          key="0"
          onClick={[MockFunction]}
          tabIndex="-1"
          title="Work Order"
          value={0}
        >
          Work Order
        </li>
        <li
          aria-selected={true}
          className="dropdown-list-item"
          id="Item1"
          key="1"
          onClick={[MockFunction]}
          tabIndex="-1"
          title="Item1"
          value={1}
        >
          Item1
        </li>
        <li
          aria-selected={false}
          className="dropdown-list-item"
          id="Item2"
          key="2"
          onClick={[MockFunction]}
          tabIndex="-1"
          title="Item2"
          value={2}
        >
          Item2
        </li>
        <li
          aria-selected={false}
          className="dropdown-list-item"
          id="Item3"
          key="3"
          onClick={[MockFunction]}
          tabIndex="-1"
          title="Item3"
          value={3}
        >
          Item3
        </li>
      </ul>
    </div>
  </div>
</DropdownField>
`;
